Source/Initialization/GetTemperature.H

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2021 Hannah Klion
+ *
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef GET_TEMPERATURE_H_
+#define GET_TEMPERATURE_H_
+
+#include "TemperatureProperties.H"
+
+/**
+ * \brief Get temperature at a point on the grid
+ *
+ * Functor to return temperature at a point on the grid, either constant (m_temperature)
+ * or a spatially varying value computed using the parser function (m_temperature_parser).
+ * It provides the temperature information held by the TemperatureProperties instance
+ * passed to the constructor.
+ */
+struct GetTemperature
+{
+    /* Type of temperature initialization */
+    TemperatureInitType m_type;
+
+    /* Constant temperature value, if m_type == ConstantValue */
+    amrex::Real m_temperature;
+    /* Temperature parser function, if m_type == ParserFunction */
+    amrex::ParserExecutor<3> m_temperature_parser;
+
+    /**
+     * \brief Construct the functor with information provided by temp
+     *
+     * \param[in] temp: const reference to the TemperatureProperties object that will be used to
+     * populate the functor
+     */
+    GetTemperature (TemperatureProperties const& temp) noexcept;
+
+    /**
+     * \brief Functor call. Returns the value of temperature at the location (x,y,z)
+     *
+     * \param[in] x: x-coordinate of given location
+     * \param[in] y: y-coordinate of given location
+     * \param[in] z: z-cooridnate of given location
+     *
+     *\return value of temperature at (x,y,z).
+     *        m_temperature if m_type is ConstantValue
+     *        m_temperature_parser(x,y,z) if m_type is ParserFunction
+     */
+    AMREX_GPU_HOST_DEVICE
+    amrex::Real operator() (amrex::Real x, amrex::Real y, amrex::Real z) const noexcept
```

Comment:
The `x`, `y` and `z` in this interface can be declared `const` :)

Reply:
Thanks for catching that! I've pushed a change.

I've run small by-eye checks in 2D and 3D with CUDA on my local machine, plus the CI compile checks. Is there anything else I should do?